### PR TITLE
Documentation updated to support file creation using metadata trigger…

### PR DESCRIPTION
**Pull request by datamel**
…. Test also included.
This refers to Issue #2300 .

### CODE REVIEW

**Comment by oliver-sanders** [comment]
@metomi/core could someone eyeball this :eye: .

File: sphinx/api/configuration/metadata.rst [hunk 1]
```diff
@@ -685,6 +685,40 @@ The metadata options for a configuration fall into four categories:
          expression syntax can be used with non-numeric variables in the same
          way as the fail-if metadata.
 
+         .. _trigger-file-creation:
+
+         It is possible to use metadata triggers to trigger file creation, 
+         switching on/off as required.
+
+         For example, with the following rose-app.conf and rose-meta.conf
```
Comment (sadielbartholomew) [inline sphinx/api/configuration/metadata.rst]: Not a blocker, but for clarity & consistency it is nice to keep literal identifiers such as file names in monospace font, i.e:

```suggestion
         For example, with the following ``rose-app.conf`` and ``rose-meta.conf``
```

File: sphinx/api/configuration/metadata.rst [hunk 2]
```diff
@@ -590,7 +590,7 @@ The metadata options for a configuration fall into four categories:
 
          * normal
          * user ignored (stored in the configuration file with a ``!`` flag,
-           ignored at run time)
+           ignored at runtime)
```
Comment (sadielbartholomew) [inline sphinx/api/configuration/metadata.rst]: Good spot!

File: sphinx/api/configuration/metadata.rst (hunk 1 above)
Comment (oliver-sanders) [reply sphinx/api/configuration/metadata.rst]: Or better still:

```rst
:rose:file:`rose-app.conf`
```

Comment (datamel) [reply sphinx/api/configuration/metadata.rst]: Thanks! I'll change this now :)

**Comment by oliver-sanders** [comment]
@datamel 

Sorry, forgot to mention GitHub has a magic feature where if you write something like this in a pull request description:

```
Closes #1234 
```

GitHub will automatically close the issue 1234 when the PR is merged.

**Comment by datamel** [comment]
Thanks, good to know - if I remember ;)

**Pull request merged**